nemo_skills/mcp/clients.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import abstractmethod, ABC
+import aiohttp
+import functools
+from typing import Dict, Any, List
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.streamable_http import streamablehttp_client
+from mcp.client.stdio import stdio_client
+
+import copy
+
+def _process_hide_args(result, hide_args):
+    if hide_args:
+        output = []
+        for entry in result:
+            method_name = entry.get('name')
+            schema = copy.deepcopy(entry.get('input_schema', {}))
+            if method_name in hide_args and 'properties' in schema:
+                for arg in hide_args[method_name]:
+                    schema['properties'].pop(arg, None)
+                    if 'required' in schema and arg in schema['required']:
+                        schema['required'].remove(arg)
+            new_entry = dict(entry)
+            new_entry['input_schema'] = schema
+            output.append(new_entry)
+        return output
+    return result
+
+
+def async_wrapper(method):
+    async def wrapped(self, *args, **kwargs):
+        hide_args = kwargs.pop('hide_args', None)
+        if hide_args is None:
+            hide_args = getattr(self, '_hide_args', {})
+        result = await method(self, *args, **kwargs)
+        return _process_hide_args(result, hide_args)
+    return wrapped
+
+def inject_hide_args(init_func):
+    @functools.wraps(init_func)
+    def wrapper(self, *args, hide_args=None, **kwargs):
+        self._hide_args = hide_args or {}
+        return init_func(self, *args, **kwargs)
+    return wrapper
+
+class MCPClientMeta(type):
```

Comment:
can you please explain what's the purpose of this metaclass?

Reply:
okay I've added some more functionality here as well as a docstring that explains it